### PR TITLE
PREAPPS-4167: Inessential requests get fired on navigate to Calendar Tab

### PR DESCRIPTION
**Pull request by SantoshKShetty**

### CODE REVIEW

File: src/batch-client/index.ts [hunk 1]
```diff
@@ -882,6 +882,27 @@ export class ZimbraBatchClient {
 			namespace: Namespace.Mail
 		}).then(({ tag = [] }) => tag.map(normalize(Tag)));
 
+	public getTasks = (options: SearchOptions) =>
```
Comment (silentsakky) [inline src/batch-client/index.ts]: didn't we already have query to get task data?
and also we should map query with soap api endpoint

Comment (SantoshKShetty) [reply src/batch-client/index.ts]: Hmm.. It seems so.. Need to find out why new query was used.

Comment (SantoshKShetty) [reply src/batch-client/index.ts]: @silentsakky there's a TaskQuery but its different (being used to get task data for given task). This is a search task request.

Comment (itsmeprashant) [reply src/batch-client/index.ts]: We are doing same thing here: 
https://github.com/Zimbra/zm-api-js-client/blob/develop/src/schema/schema.ts#L220

Then why this chenage is needed?

File: src/schema/schema.ts [hunk 2]
```diff
@@ -199,41 +197,6 @@ export function createZimbraSchema(
 					return obj.conversationId ? 'MessageInfo' : 'Conversation';
 				}
 			},
-			Folder: {
```
Comment (itsmeprashant) [inline src/schema/schema.ts]: Why did we remove this?

Comment (SantoshKShetty) [reply src/schema/schema.ts]: Found no usage of it.

Comment (yashrajpchavda) [reply src/schema/schema.ts]: @SantoshKShetty should we remove the respective `folder -> appointments` mapping from schema.graphql too?

Comment (SantoshKShetty) [reply src/schema/schema.ts]: Nice catch. Fixed.

Comment (SantoshKShetty) [reply src/schema/schema.ts]: For now, I have removed only `folder -> tasks` mapping from schema.graphql as we're still using CalendarsAndAppointmentsQuery in new-event-from-email zimlet.

Once we refactor that zimlet, we'll update schema.graphql and remove `folder -> appointments`

**Pull request merged**